### ISSUE TITLE
UnnecessaryDefaultInEnumSwitch doesn't consider return statements.

### ISSUE THREAD

**Issue by hzsweers** [google/error-prone#548]
The following snippet would get flagged by the `UnnecessaryDefaultInEnumSwitch` check, and it suggests removing the `default` line. Thing is, removing that is actually a compile error because of a perceived missing return.

```java
private static final Func1<ActivityEvent, ActivityEvent> ACTIVITY_LIFECYCLE =
        new Func1<ActivityEvent, ActivityEvent>() {
            @Override
            public ActivityEvent call(ActivityEvent lastEvent) {
                switch (lastEvent) {
                    case CREATE:
                        return ActivityEvent.DESTROY;
                    case START:
                        return ActivityEvent.STOP;
                    case RESUME:
                        return ActivityEvent.PAUSE;
                    case PAUSE:
                        return ActivityEvent.STOP;
                    case STOP:
                        return ActivityEvent.DESTROY;
                    case DESTROY:
                        throw new OutsideLifecycleException("Cannot bind to Activity lifecycle when outside of it.");
                    default:
                        throw new UnsupportedOperationException("Binding to " + lastEvent + " not yet implemented");
                }
            }
        };
```

**Comment by cushon** [comment]
This is an issue with the way suggested fixes that span multiple lines are formatted in diagnostics.

The full suggested fix is to replace this:

```java
                    case DESTROY:
                        throw new OutsideLifecycleException("Cannot bind to Activity lifecycle when outside of it.");
                    default:
                        throw new UnsupportedOperationException("Binding to " + lastEvent + " not yet implemented");
                }
            }
```

with this:

```java
                    case DESTROY:
                        throw new OutsideLifecycleException("Cannot bind to Activity lifecycle when outside of it.");
                }
                throw new UnsupportedOperationException("Binding to " + lastEvent + " not yet implemented");
            }
```

**Comment by hzsweers** [comment]
Ahh gotcha. Thanks!

**Comment by cushon** [comment]
The diagnostic message should still be improved, I filed #551 for that.
